File: testers/avro/java-async/generated-and-checked-in/com/example/service/UserService.java
package com.example.service;

import java.util.concurrent.CompletableFuture;

/** User management service protocol */
public interface UserService {
  /** Get a user by their ID */
  CompletableFuture<Result<User, UserNotFoundError>> getUser(String userId);

  /** Create a new user */
  CompletableFuture<Result<User, ValidationError>> createUser(String email, String name);

  /** Delete a user */
  CompletableFuture<Result<Void, UserNotFoundError>> deleteUser(String userId);

  /** Send a notification to a user (fire-and-forget) */
  CompletableFuture<Void> notifyUser(String userId, String message);
}

File: testers/avro/kotlin-json/generated-and-checked-in/com/example/service/UserService.kt
package com.example.service



/** User management service protocol */
interface UserService {
  /** Create a new user */
  abstract fun createUser(
    email: kotlin.String,
    name: kotlin.String
  ): Result<User, ValidationError>

  /** Delete a user */
  abstract fun deleteUser(userId: kotlin.String): Result<Unit, UserNotFoundError>

  /** Get a user by their ID */
  abstract fun getUser(userId: kotlin.String): Result<User, UserNotFoundError>

  /** Send a notification to a user (fire-and-forget) */
  abstract fun notifyUser(
    userId: kotlin.String,
    message: kotlin.String
  )
}